Account/FxAPushMessageHandler.swift

```diff
@@ -60,76 +60,83 @@ extension FxAPushMessageHandler {
         }
 
         // return handle(plaintext: string)
-        let deferred = PushMessageResult()
+        let deferred = PushMessageResults()
         RustFirefoxAccounts.reconfig(prefs: profile.prefs).uponQueue(.main) { accountManager in
             accountManager.deviceConstellation()?.processRawIncomingAccountEvent(pushPayload: string) {
                 result in
-                guard case .success(let events) = result, let firstEvent = events.first else {
+                guard case .success(let events) = result, !events.isEmpty else {
                     let err: PushMessageError
                     if case .failure(let error) = result {
+                        SentryIntegration.shared.send(message: "Failed to get any events from FxA", tag: .fxaClient, severity: .error, description: error.localizedDescription)
                         err = PushMessageError.messageIncomplete(error.localizedDescription)
                     } else {
+                        SentryIntegration.shared.send(message: "Got zero events from FxA", tag: .fxaClient, severity: .error, description: "no events retrieved from fxa")
                         err = PushMessageError.messageIncomplete("empty message")
                     }
                     deferred.fill(Maybe(failure: err))
                     return
                 }
-                if events.count > 1 {
-                    // Log to the console for debugging release builds
-                    os_log(
-                        "%{public}@",
-                        log: OSLog(subsystem: "org.mozilla.firefox",
-                                   category: "firefoxnotificationservice"),
-                        type: OSLogType.debug,
-                        "Multiple events arrived, only handling the first event.")
-                }
-                switch firstEvent {
-                case .commandReceived(let deviceCommand):
-                    switch deviceCommand {
-                    case .tabReceived(_, let tabData):
-                        let title = tabData.entries.last?.title ?? ""
-                        let url = tabData.entries.last?.url ?? ""
-                        let message = PushMessage.commandReceived(tab: ["title": title, "url": url])
-                        if let json = try? accountManager.gatherTelemetry() {
-                            let events = FxATelemetry.parseTelemetry(fromJSONString: json)
-                            events.forEach { $0.record(intoPrefs: self.profile.prefs) }
+                var messages: [PushMessage] = []
+
+                // It's possible one of the messages is a device disconnection
+                // in that case, we have an async call to get the name of the device
+                // we should make sure not to resolve our own value before that name retrieval
+                // is done
+                var waitForClient: Deferred<Maybe<String>>?
```

Comment:
This is hacky, and, I think device disconnection is broken in production.

From my local testing:
- when disconnecting a firefox desktop, the device does **not** even get a notification!
- when disconnecting another firefox iOS app, the device gets the notification but doesn't display the device's name correctly. Regardless that may be  something we should fix so keeping the logic we have here